src/enum.lisp

```diff
@@ -37,59 +47,138 @@
 ;;; constant for the type by a type translator when passed as
 ;;; arguments or a return value to a foreign function.
 
-(defclass foreign-enum (foreign-typedef enhanced-foreign-type)
+(defclass foreign-enum (named-foreign-type enhanced-foreign-type)
```

Comment:
who remembers after such a long time...? :)

i think the problem was that ```follow-typedefs``` went all the way down to :int somewhere, which was wrong. an enum is not a typedef to an int, it's an enum...